tygo/write.go

```diff
@@ -36,13 +36,13 @@ func (g *PackageGenerator) writeIndent(s *strings.Builder, depth int) {
 	}
 }
 
-func (g *PackageGenerator) writeType(s *strings.Builder, t ast.Expr, depth int, optionalParens bool) {
+func (g *PackageGenerator) writeType(s *strings.Builder, t ast.Expr, depth int, optionalParens bool, preserveComments bool) {
```

Comment:
Nit, not a blocker, feel free to disagree if you don't prefer it:

Instead of passing it down, you could access `g.conf.PreserveComments`, that will save you quite some plumbing in this PR.

I would propose to add a helper method to `*PackageConfig` or even to `*PackageGenerator`:
```go
func (g *PackageGenerator) PreserveTypeComments() bool {
  return g.conf.PreserveComments == "types" || g.conf.PreserveComments == "default"
}
```

(Or you could even add a `PreserveTypeComments`, `PreserveGroupComments`, ... bool fields)